### PR TITLE
[bugfix](paimon)add class loader

### PR DESCRIPTION
**Pull request by wuwenchi**
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wuwenchi** [comment]
run buildall

File: fe/be-java-extensions/paimon-scanner/src/main/java/org/apache/doris/paimon/PaimonJniScanner.java [hunk 1]
```diff
@@ -87,9 +89,18 @@ public PaimonJniScanner(int batchSize, Map<String, String> params) {
 
     @Override
     public void open() throws IOException {
-        initTable();
-        initReader();
-        resetDatetimeV2Precision();
+        try {
+            Thread.currentThread().setContextClassLoader(classLoader);
```
Comment (morningman) [inline fe/be-java-extensions/paimon-scanner/src/main/java/org/apache/doris/paimon/PaimonJniScanner.java:93]: Add comment in code to explain why need class loader

Comment (wuwenchi) [reply fe/be-java-extensions/paimon-scanner/src/main/java/org/apache/doris/paimon/PaimonJniScanner.java:97]: done

File: fe/be-java-extensions/paimon-scanner/src/main/java/org/apache/doris/paimon/PaimonJniScanner.java [hunk 2]
```diff
@@ -87,9 +89,18 @@ public PaimonJniScanner(int batchSize, Map<String, String> params) {
...
+            initTable();
+            initReader();
+            resetDatetimeV2Precision();
+        } catch (Exception e) {
+            LOG.warn("Failed to open paimon_scanner: " + e.getMessage());
```
[the remark this answers is not in the document]
Comment (wuwenchi) [reply fe/be-java-extensions/paimon-scanner/src/main/java/org/apache/doris/paimon/PaimonJniScanner.java]: done

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Pull request merged**